xla/tests/convert_test.cc

```diff
@@ -752,8 +775,18 @@ XLA_TEST_F(ConvertTest, ConvertF16F8e5m2Roundtrip) {
       {0x1.DFCp15, 0x1.Cp15},  // Largest number that doesn't overflow
       {0x1.Ep15, inf},         // Smallest number that overflows
       {0x1p16, inf},           // Overflow
-      {0x1p-14, 0x1p-14},      // Smallest normal
-      {0x1.8p-15, 0x1.8p-15},  // Denormal
+      {0x1p-14, 0x1p-14},      // Smallest F8 normal
+      {0x1.Cp-15, 0x1p-14},    // Smallest number rounding up to normal
+
+      // Denormal tests
+      {0x1.0p-15, 0x1.0p-15},   // Denormal without rounding
+      {0x1.4p-15, 0x1.0p-15},   // Round-to-even down
+      {0x1.Cp-15, 0x1.0p-14},   // Round-to-even up
+      {0x1.3p-15, 0x1.0p-15},   // Round-to-nearest down
+      {0x1.5p-15, 0x1.8p-15},   // Round-to-nearest up
+      {0x1p-17, 0},             // Largest number that underflows
+      {0x1.04p-17, 0x1p-16},    // Smallest number that doesn't underflow
```

Comment:
I think you wanted to pass the following input for "Smallest number that doesn't underflow"
```
Input: 0.00000.0010 0000 01
Expected: 0.00000.0100 0000 00   # min subnormal
```
If you use `0x0.` denorm notation the you can easily convert the binary numbers above to hex
```
Input: 0.00000.0010 0000 01
mantisa 0010 0000 0100 = 0x0.204
exponent 00000 =  p-14 (denorm e5m2)
0x0.204p-14

Expected: 0.00000.0100 0000 00   # min subnormal
mantisa 0100 0000 0000 = 0x0.400
exponent 00000 =  p-14 (denorm e5m2)
0x0.4p-14
```

The correct Smallest number that doesn't underflow is `0x0.204p-14` which is smaller than `0x1.04p-17`
```
## Fixed test
{0x0.204p-14, 0x0.4p-14}, // Smallest number that doesn't underflow
```